File: Populating Next Right Pointers in Each Node/Populating Next Right Pointers in Each Node.java
/**
 * Definition for binary tree with next pointer.
 * public class TreeLinkNode {
 *     int val;
 *     TreeLinkNode left, right, next;
 *     TreeLinkNode(int x) { val = x; }
 * }
 */
public class Solution {
    public void connect(TreeLinkNode root) {
        TreeLinkNode left=root;
        while(left!=null){
            TreeLinkNode across=left;
            while(across!=null){
                if(across.left!=null){
                    across.left.next=across.right;
                }
                if(across.right!=null&&across.next!=null){
                    across.right.next=across.next.left;
                }
                across=across.next;
            }
            left=left.left;
        }
    }
}

File: Populating Next Right Pointers in Each Node/Populating Next Right Pointers in Each Node.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
#         self.next = None

class Solution:
    # @param root, a tree node
    # @return nothing
    def connect(self, root):
        left=root
        while left:
            across=left
            while across:
                if across.left:
                    across.left.next=across.right
                if across.right and across.next:
                    across.right.next=across.next.left
                across=across.next
            left=left.left
            
